generate_emoji_name_data.py

```diff
@@ -0,0 +1,401 @@
+  We opt to simplify the former to an activity name or action, and the latter to
+  drop the gender.  This also generally makes the names shorter."""
+
+  EYE = 0x1f441
+  SPEECH = 0x1f5e8
+  WHITE_FLAG = 0x1f3f3
+  RAINBOW = 0x1f308
+  return {
+      (EYE, SPEECH): 'I Witness',
+      (WHITE_FLAG, RAINBOW): 'Rainbow Flag',
+      (0x2695,): 'Health Worker',
+      (0x2696,): 'Judge',
+      (0x26f7,): 'Skiing',
+      (0x26f9,): 'Bouncing a Ball',
+      (0x2708,): 'Pilot',
+      (0x1f33e,): 'Farmer',
+      (0x1f373,): 'Cook',
+      (0x1f393,): 'Student',
+      (0x1f3a4,): 'Singer',
+      (0x1f3a8,): 'Artist',
+      (0x1f3c2,): 'Snowboarding',
+      (0x1f3c3,): 'Running',
+      (0x1f3c4,): 'Surfing',
+      (0x1f3ca,): 'Swimming',
+      (0x1f3cb,): 'Weight Lifting',
+      (0x1f3cc,): 'Golfing',
+      (0x1f3eb,): 'Teacher',
+      (0x1f3ed,): 'Factory Worker',
+      (0x1f46e,): 'Police Officer',
+      (0x1f46f,): 'Partying',
+      (0x1f471,): 'Person with Blond Hair',
+      (0x1f473,): 'Person Wearing Turban',
+      (0x1f477,): 'Construction Worker',
+      (0x1f481,): 'Tipping Hand',
+      (0x1f482,): 'Guard',
+      (0x1f486,): 'Face Massage',
+      (0x1f487,): 'Haircut',
+      (0x1f4bb,): 'Technologist',
+      (0x1f4bc,): 'Office Worker',
+      (0x1f527,): 'Mechanic',
+      (0x1f52c,): 'Scientist',
+      (0x1f575,): 'Detective',
+      (0x1f645,): 'No Good Gesture',
+      (0x1f646,): 'OK Gesture',
+      (0x1f647,): 'Bowing Deeply',
+      (0x1f64b,): 'Raising Hand',
+      (0x1f64d,): 'Frowning',
+      (0x1f64e,): 'Pouting',
+      (0x1f680,): 'Astronaut',
+      (0x1f692,): 'Firefighter',
+      (0x1f6a3,): 'Rowing',
+      (0x1f6b4,): 'Bicycling',
+      (0x1f6b5,): 'Mountain Biking',
+      (0x1f6b6,): 'Walking',
+      (0x1f926,): 'Face Palm',
+      (0x1f937,): 'Shrug',
+      (0x1f938,): 'Doing a Cartwheel',
+      (0x1f939,): 'Juggling',
+      (0x1f93c,): 'Wrestling',
+      (0x1f93d,): 'Water Polo',
+      (0x1f93e,): 'Playing Handball',
+      (0x1f9d6,): 'Person in Steamy Room',
+      (0x1f9d7,): 'Climbing',
+      (0x1f9d8,): 'Person in Lotus Position',
+      (0x1f9d9,): 'Mage',
+      (0x1f9da,): 'Fairy',
+      (0x1f9db,): 'Vampire',
+      (0x1f9dd,): 'Elf',
+      (0x1f9de,): 'Genie',
+      (0x1f9df,): 'Zombie',
+  }
+
+_CUSTOM_GENDERED_SEQ_NAMES = _create_custom_gendered_seq_names()
+_CUSTOM_SEQ_NAMES = _create_custom_seq_names()
+
+# Fixes for unusual capitalization or cases we don't care to handle in code.
+# Also prevents titlecasing 'S' after apostrophe in posessives.  Note we _do_
+# want titlecasing after apostrophe in some cases, e.g. O'Clock.
+_CUSTOM_CAPS_NAMES = {
+    (0x26d1,): 'Rescue Worker’s Helmet',
+    (0x1f170,): 'A Button (blood type)',  # a Button (Blood Type)
+    (0x1f171,): 'B Button (blood type)',  # B Button (Blood Type)
+    (0x1f17e,): 'O Button (blood type)',  # O Button (Blood Type)
+    (0x1f18e,): 'AB Button (blood type)',  # Ab Button (Blood Type)
```

Comment:
Not only that, why are these emoji?  And why is 'AB' default emoji presentation, but the others not? And why, in the emoji groupings, are these tossed in together with with other 'text' emoji and listed alphabetically, which splits 'O' away from the others?...  so many questions, so little time.